simple_ldap_user/simple_ldap_user.module

```diff
@@ -481,7 +481,16 @@ function simple_ldap_user_sync_user_to_drupal($drupal_user) {
   $ldap_user = SimpleLdapUser::singleton($drupal_user->name, TRUE);
 
   // Nothing to sync.
-  if (!$ldap_user->exists || (isset($drupal_user->uid) && $drupal_user->uid == 1)) {
+  if(isset($drupal_user->uid) && $drupal_user->uid == 1) {
+    // Check if user is an admin.
+    return;
+  } elseif (!$ldap_user->exists) {	//For other users.
+    $result = db_update('users')
+              ->fields(array(
+                  'status' => 0
+              )) //Update status to 0 for blocked users
+              ->condition('name',$drupal_user->name,'=')
```

Comment:
Generally, use the uid as the key for an update like this.  The `$user->name` should be a unique key, but `$user->uid` is the table's primary key.